components/brave_wallet_ui/common/slices/api.slice.ts

```diff
@@ -746,8 +748,25 @@ export function createWalletApi (
                 'braveWalletService.SetChainIdForActiveOrigin failed'
               )
             }
+
+            // FIXME(josheleonard): could be written in a more efficient way.
+            // Clients setting networks must be aware of the selected account
+            // corresponding to the new network.
+            const accountsRegistry = await cache.getAccountsRegistry()
+            cache.clearSelectedAccount()
+            const selectedAccountAddress =
+              await cache.getSelectedAccountAddress()
+
+            const selectedAccount = selectedAccountAddress
+              ? accountsRegistry.entities[
+                  accountInfoEntityAdaptor.selectId({
+                    address: selectedAccountAddress
+                  })
+                ]
+              : undefined
```

Comment:
This mutation invalidates the selected account cache. any client wishing to observe the update to the new account should utilize the `useSelectedAccountQuery`  or `useGetSelectedAccountAddress` hooks